96752745_Amal/assignment3.cu

```diff
@@ -0,0 +1,53 @@
+#include<iostream>
+using namespace std;
+
+__global__ void Transpose(int *d_a,int max){
+
+int i = blockIdx.x*blockDim.x+threadIdx.x;
+int j = blockIdx.y*blockDim.y+threadIdx.y;
+int id1 = i+max*j;
+int id2 = j+max*i;
+__syncthreads();
+
+if(i<max && j<max)
+{
+	int t = d_a[id1];
+	__syncthreads();
+	d_a[id1]=d_a[id2];

```

Comment:
This line is redundant. 
```
int t = d_a[id1];
__syncthreads();
d_a[id2] = t;
```
This much itself should do the job, because you are assigning a thread to every matrix element and so every matrix index will get written by its particular thread. You do not need to write to two locations in a single thread.